client/app/scripts/liveblog-edit/directives.js

```diff
@@ -50,40 +50,116 @@ define([
                 }
             };
         })
-        .directive('draftPosts', ['api', 'postsService',
-            function DraftPostsDirective(api, postsService) {
-                DraftPostsController.$inject = ['$scope'];
-                function DraftPostsController($scope) {
-                    var mv = this;
-                    mv.posts = [];
-                    mv.postsInfo = {};
-                    mv.selectDraftPost = $scope.onDraftPostSeleted;
-                    function updateList() {
-                        postsService.fetchDrafts(
-                            $scope.blog._id,
-                            mv.posts,
-                            mv.postsInfo
-                        );
+        .directive('lbPostsList', [
+            'postsService', 'notify', '$q',
+            function(postsService, notify, $q) {
+
+                LbPostsListCtrl.$inject = ['$scope'];
+                function LbPostsListCtrl($scope) {
+                    var vm = this;
+
+                    function fetchPage() {
+                        // stop if the limit is reached
+                        if (angular.isDefined(vm.postsMeta.total) && vm.postsMeta.total <= vm.posts.length) {
+                            return false;
+                        }
+                        // Find the next page containing new posts
+                        var page = 1;
+                        while (vm.pagination.limit * page <= vm.posts.length) {
```

Comment:
I also changed the update mechanism, it is now totally separated from the pagination.
If I receive 7 updates, anyway, if I need a new page the while will find the next one containing new posts. Maybe it won't contains the maximum number of posts, but it's not an issue in an infinite scroll interface. Just get down to have more :)